doc/source/whatsnew/v0.18.1.txt

```diff
@@ -428,3 +428,4 @@ Bug Fixes
 
 
 - Bug in ``fill_value`` is ignored if the argument to a binary operator is a constant (:issue `12723`)
+- Bug in ``DataFrame._santize_column`` now preserves Timezone when RHS is a Series (:issue `12981`)
```

Comment:
say setting a column with alignment to a ``Series`` that is datetime tz-aware dtype. We don't expect readers to know about internal functions, just the outward effects.